python/cudf/cudf/core/multiindex.py

```diff
@@ -1733,3 +1733,33 @@ def _intersection(self, other, sort=None):
         if sort is None and len(other):
             return midx.sort_values()
         return midx
+
+    def _split_columns_by_levels(self, levels):
+        # This function assumes that for levels with duplicate names, they are
+        # specified by indices, not name by ``levels``. E.g. [None, None] can
+        # only be specified by 0, 1, not "None".
+
+        # Normalize named levels into indices
+        if levels is not None:
+            level_indices = set()
+            level_names = list(self.names)
+            for lv in levels:
+                if isinstance(lv, int):
+                    level_indices.add(lv)
+                else:
+                    level_indices.add(level_names.index(lv))
```

Comment:
This suggestion will probably break black's line limit and require reformatting.
```suggestion
            level_names = list(self.names)
            level_indices = {lv if isinstance(lv, int) else level_names.index(lv) for lv in levels}
```